examples/tools/file_reader.py

```diff
@@ -1,74 +1,133 @@
 
-class FileReadTool(Node):
+
+class FileReaderTool(Node):
     """
-    A tool to read the content of a file.
+    A tool to read the content of one or more files from byte streams.
 
     Attributes:
         name (str): The name of the tool.
         description (str): The description of the tool.
-        file_path (str): The file path to read.
+        files (list[bytes | io.BytesIO] | None): List of files to read content from.
+        show_partial (bool): Whether to show only a part of the content to avoid overwhelming the output.
+        max_display_length (int): Maximum number of characters/bytes to display in the output.
     """
 
     group: Literal[NodeGroup.TOOLS] = NodeGroup.TOOLS
-    name: str = "Read a file's content"
-    description: str = "A tool that can be used to read a file's content from local storage."
-    file_path: str
+    name: str = "file-reader-tool"
+    description: str = DESCRIPTION
+    files: list[bytes | io.BytesIO] | None = None
+    show_partial: bool = True
+    max_display_length: int = 500
+    supports_files: bool = True
 
     model_config = ConfigDict(arbitrary_types_allowed=True)
 
-    def _read_file_content(self, file_path: str) -> Any:
+    def _load_file_content(self, file: bytes | io.BytesIO) -> str:
         """
-        Read the content of the file at the given path.
+        Load the content of a file from bytes or a byte stream.
 
         Args:
-            file_path (str): The path to the file.
+            file (Union[bytes, io.BytesIO]): The byte content or stream of the file.
 
         Returns:
-            Any: The content of the file.
+            str: The loaded content, either fully or partially (based on configuration).
         """
         try:
-            with open(file_path) as file:
-                return file.read()
+            file_description = getattr(file, "description", "Unnamed file")
+
+            logger.debug(f"Reading file: {file_description}")
+
+            if isinstance(file, bytes):
+                content = file
+
+            elif isinstance(file, io.BytesIO):
+                file.seek(0)
+                content = file.read()
+
+            else:
+                raise ToolExecutionException(
+                    f"Invalid input type. Expected bytes or BytesIO, got {type(file)}", recoverable=False
+                )
+
+            return self._get_partial_content(content)
+
         except Exception as e:
-            logger.error(
-                f"Tool {self.name} - {self.id}: failed to to read the file {file_path}. Error: {e}"
-            )
-            raise ToolExecutionException(
-                f"Failed to to read the file {file_path}. Error: {e}", recoverable=True
-            )
+            logger.error(f"Failed to read file: {file_description}. Error: {e}")
+            raise ToolExecutionException(f"Failed to read the file: {file_description}. Error: {e}", recoverable=True)
+
+    def _get_partial_content(self, content: bytes) -> str:
```

Comment:
I don't understand the purpose of this partial functionality. You are tying to read the file partially for which reason?

Reply:
In case of very big files, just from safety purposes

Reply:
But we cut by default by 500 bytes. Is this ok?
Let's maybe allow any (or some bigger number), so user could configure by self if needs